logisland-components/logisland-processors/logisland-processor-xml/src/main/java/com/hurence/logisland/processor/xml/EvaluateXPath.java

```diff
@@ -0,0 +1,280 @@
+import java.util.concurrent.atomic.AtomicReference;
+
+import static javax.xml.xpath.XPathConstants.STRING;
+
+
+@Tags({"XML", "evaluate", "XPath"})
+@CapabilityDescription("Evaluates one or more XPaths against the content of a record. The results of those XPaths are assigned to "
+        + "new attributes in the records, depending on configuration of the "
+        + "Processor. XPaths are entered by adding user-defined properties; the name of the property maps to the Attribute "
+        + "Name into which the result will be placed. "
+        + "The value of the property must be a valid XPath expression. If the expression matches nothing, no attributes is added. ")
+@WritesAttribute(attribute = "user-defined", description = "This processor adds user-defined attributes.")
+@DynamicProperty(name = "An attribute", value = "An XPath expression", description = " "
+        + "the attribute is set to the result of the XPath Expression.")
+
+public class EvaluateXPath extends AbstractProcessor {
+
+    protected HashMap<String, XPathExpression> xpathRules = new HashMap<String, XPathExpression>();
+
+    public static final PropertyDescriptor SOURCE = new PropertyDescriptor.Builder()
+            .name("Source")
+            .description("Indicates the attribute containing the xml data to evaluate xpath against.")
+            .required(true)
+            .build();
+
+    public static final PropertyDescriptor VALIDATE_DTD = new PropertyDescriptor.Builder()
+            .name("Validate DTD")
+            .description("Specifies whether or not the XML content should be validated against the DTD.")
+            .required(true)
+            .allowableValues("true", "false")
+            .defaultValue("true")
+            .build();
+
+
+    private static final AllowableValue OVERWRITE_EXISTING =
+            new AllowableValue("overwrite_existing", "overwrite existing field", "if field already exist");
+
+    private static final AllowableValue KEEP_OLD_FIELD =
+            new AllowableValue("keep_only_old_field", "keep only old field value", "keep only old field");
+
+
+    public static final PropertyDescriptor CONFLICT_RESOLUTION_POLICY = new PropertyDescriptor.Builder()
+            .name("conflict.resolution.policy")
+            .description("What to do when a field with the same name already exists ?")
+            .required(false)
+            .defaultValue(KEEP_OLD_FIELD.getValue())
+            .allowableValues(OVERWRITE_EXISTING, KEEP_OLD_FIELD)
+            .build();
+
+
+
+    private List<PropertyDescriptor> properties;
+
+    private final AtomicReference<XPathFactory> factoryRef = new AtomicReference<>();
+
+    static {
+        System.setProperty("javax.xml.xpath.XPathFactory:" + NamespaceConstant.OBJECT_MODEL_SAXON, "net.sf.saxon.xpath.XPathFactoryImpl");
+    }
+
+    @Override
+    public void init(ProcessContext context) {
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(SOURCE);
```

Comment:
Please construct the supported properties in the constructor (or just in the getSupportedPropertyDescriptors method). Otherwise the automatic documentation risks not to work for this processor.